### DISCUSSION TITLE
get last version tag from git

### DISCUSSION THREAD

**Discussion by IlanCosman** [fish-shell/fish-shell#8397] [Q&A] [answered]
How would I `read` the last line of input into a variable?

Example:
```fish
ilan@arch ~/tide ((v5))> git tag --points-at HEAD
v5
v5.0
v5.0.1
```

I'd like to read in `v5.0.1` instead of `v5`.

**Comment by krobelus** [comment]
`set version (git tag --sort=version:refname | sed -n '$p')`

The default lexicographic sorting of git tag will go wrong if there are versions like `0.1.0` and `0.10.0`

**Reply by IlanCosman** [reply]
Oh god I did the [xy problem](https://xyproblem.info/) 😞
I think I'll just stick with the first tag then, I don't want to call external programs.
Thank you @krobelus!

**Reply by faho** [reply]
This also fails on the fish-shell repo because we have some tags that fail a version sort - like "pre_whitespace_fix". They will be sorted as larger than the largest version.

Sorting by taggerdate is another idea that won't work because of multiple version tracks - if we released a 3.2.3 now it would pick that instead of 3.3.1. Just like sorting by number of commits (which would require git-for-each-ref trickery IIUC) - theoretically 3.2.3 could contain more commits than 3.3.1.

The best I can come up with is `git describe --tags | string replace -r -- '-\d+-g[0-9a-f]+$' ''` (note `--tags` is a misnomer - it tells it to use *all tags*, not just annotated ones. The idea is that you'd annotate release tags but not informative ones like pre_whitespace_fix. I don't know if that happens in practice).

`git describe` picks the *closest* tag to your HEAD (maybe use `git describe --tags master` if you haven't checked that out yet), so it does the heavy lifting. We then just need to remove the guff it adds if it's not an exact match. If you are only interested in exact matches, use `git describe --exact-match`

**Comment by krobelus** [comment, marked as answer]
> I think I'll just stick with the first tag then,

you can also use reverse sorting with a leading dash (`git tag --sort=-version:refname`),
or use `set version (git tag)[-1]` etc.

> I don't want to call external programs.

:(
